app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt

```diff
@@ -103,7 +103,7 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
       bookDelegate,
       BookOnDiskDelegate.LanguageDelegate
     )
-    if (savedInstanceState != null) {
+    if (savedInstanceState != null && ServerUtils.isServerStarted) {
```

Comment:
Is the null check redundant?

Reply:
the bundle is used to retrieve the ip on the next line, I guess IP could be retrieved from ServerUtils? My memory of this screen has gotten a might hazy from the holidays

Reply:
Yeah @iadeelzafar I'd say you could just delete this if block and just rely on `onResume` as it is 
```
  override fun onResume() {
    super.onResume()
    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
    if (ServerUtils.isServerStarted) {
      ip = ServerUtils.getSocketAddress()
      layoutServerStarted()
    }
  }
```

Reply:
@macgills fixed.